src/HotChocolate/Core/src/Types/Types/Relay/Extensions/RelayIdFieldExtensions.cs

```diff
@@ -257,6 +257,10 @@ private static IExtendedType RewriteType(ITypeInspector typeInspector, ITypeInfo
         {
             IIdSerializer serializer =
                 completionContext.Services.GetService<IIdSerializer>() ??
+                // TODO: Possible type system memory optimization
+                // would be to singleton this IdSerializer,
+                // e.g. coalesce with ?? (_idSerializer ??= new IdSerializer())
+                // where _idSerializer is static
```

Comment:
Would you agree Michael? I can do this easily enough. Would drop a ton of instances of these guys